mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowContext.java

```diff
@@ -0,0 +1,165 @@
+  }
+
+  /**
+   * Sets the experiment to log runs to by name.
+   * @param experimentName the name of the experiment to log runs to.
+   * @throws IllegalArgumentException if the experiment name does not match an existing experiment
+   */
+  public void setExperimentName(String experimentName) throws IllegalArgumentException {
+    Optional<Experiment> experimentOpt = client.getExperimentByName(experimentName);
+    if (!experimentOpt.isPresent()) {
+      throw new IllegalArgumentException(
+        String.format("%s is not a valid experiment", experimentName));
+    }
+    experimentId = experimentOpt.get().getExperimentId();
+  }
+
+  /**
+   * Sets the experiment to log runs to by ID.
+   * @param experimentId the id of the experiment to log runs to.
+   */
+  public void setExperimentId(String experimentId) {
+    this.experimentId = experimentId;
+  }
+
+  /**
+   * Returns the experiment ID we are logging to.
+   *
+   * @return the experiment ID we are logging to.
+   */
+  public String getExperimentId() {
+    return this.experimentId;
+  }
+
+  /**
+   * Starts a MLflow run. To log data to newly created MLflow run see the methods on
+   * {@link ActiveRun}. MLflow runs should be ended using {@link ActiveRun#endRun()}
+   *
+   * @param runName The name of this run.
+   * @return An {@code ActiveRun} object to log data to.
+   */
+  public ActiveRun startRun(String runName) {
+    return startRun(runName, null);
+  }
+
+  /**
+   * Like {@link #startRun(String)} but sets the {@code mlflow.parentRunId} tag in order to create
+   * nested runs.
+   *
+   * @param runName The name of this run.
+   * @param parentRunId The ID of this run's parent
+   * @return An {@code ActiveRun} object to log data to.
+   */
+  public ActiveRun startRun(String runName, String parentRunId) {
+    Map<String, String> tags = new HashMap<>();
+    tags.put(MlflowTagConstants.RUN_NAME, runName);
+    tags.put(MlflowTagConstants.USER, System.getProperty("user.name"));
+    tags.put(MlflowTagConstants.SOURCE_TYPE, "LOCAL");
+    if (parentRunId != null) {
+      tags.put(MlflowTagConstants.PARENT_RUN_ID, parentRunId);
+    }
+
+    // Add tags from DatabricksContext if they exist
+    DatabricksContext databricksContext = DatabricksContext.createIfAvailable();
+    if (databricksContext != null) {
+      tags.putAll(databricksContext.getTags());
+    }
+
+    CreateRun.Builder createRunBuilder = CreateRun.newBuilder()
+      .setExperimentId(experimentId)
+      .setStartTime(System.currentTimeMillis());
+    for (Map.Entry<String, String> tag: tags.entrySet()) {
+      createRunBuilder.addTags(
+        RunTag.newBuilder().setKey(tag.getKey()).setValue(tag.getValue()).build());
+    }
+    RunInfo runInfo = client.createRun(createRunBuilder.build());
+
+    ActiveRun newRun = new ActiveRun(runInfo, client);
+    return newRun;
+  }
+
+  /**
+   * Like {@link #startRun(String)} but will terminate the run after the activeRunFunction is
+   * executed.
```

Comment:
Can you write example code with this? I assume there's a fancy lambda-y thing you can do in Java.

Reply:
Good idea